### PR TITLE
Firefly lora provider added

### PR DESCRIPTION
**Pull request by BobClaerhout**
Signed-off-by: Bob Claerhout <claerhout.bob@gmail.com>

### CODE REVIEW

**Comment by BobClaerhout** [comment]
Build failure seems a glitch. Can somebody trigger this agani?

**Comment by calohmn** [comment]
yes, build is green now.

File: adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/FireflyProvider.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Map;
+
+/**
+ * A LoRaWAN provider with API for Firefly.
+ */
+@Component
+public class FireflyProvider implements LoraProvider {
+
+    private static final String FIELD_FIREFLY_SPREADING_FACTOR = "spreading_factor";
+    private static final String FIELD_FIREFLY_FPORT = "port";
```
Comment (calohmn) [inline adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/FireflyProvider.java]: Did you mean `FIELD_FIREFLY_PORT`?

Comment (BobClaerhout) [reply adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/FireflyProvider.java]: No, this is the "function_port". Renamed it to `FIELD_FIREFLY_FUNCTION_PORT` to be declarative

File: adapters/lora-vertx/src/test/java/org/eclipse/hono/adapter/lora/providers/FireflyProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.lora.providers;
+
+import io.vertx.core.json.JsonObject;
+import org.eclipse.hono.adapter.lora.LoraMessageType;
+import org.junit.Assert;
+import org.junit.Test;
```
Comment (sophokles73) [inline adapters/lora-vertx/src/test/java/org/eclipse/hono/adapter/lora/providers/FireflyProviderTest.java]: please use junit5 for new test classes

Comment (BobClaerhout) [reply adapters/lora-vertx/src/test/java/org/eclipse/hono/adapter/lora/providers/FireflyProviderTest.java]: done

File: adapters/lora-vertx/pom.xml [hunk 3]
```diff
@@ -60,6 +60,12 @@
             <groupId>org.hamcrest</groupId>
             <artifactId>hamcrest-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.1.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (sophokles73) [inline adapters/lora-vertx/pom.xml]: no need for this, the parent POM already declares the jupiter dependency. Please remove, as it also overrides a *managed dependency* ...

Comment (BobClaerhout) [reply adapters/lora-vertx/pom.xml]: ok, removed

**Pull request merged**